### PR TITLE
[utils] Cross repositories pull request test support

### PR DESCRIPTION
**Pull request by shahmishal**
Add support to be able to test pull request cross repositories.

### CODE REVIEW

**Comment by shahmishal** [comment]
@swift-ci Please smoke test

**Comment by shahmishal** [comment]
@swift-ci Python lint

File: utils/update-checkout [hunk 1]
```diff
@@ -73,17 +75,18 @@ def update_single_repository(repo_path, branch, reset_to_remote, should_clean):
         # Prior to Git 2.6, this is the way to do a "git pull
         # --rebase" that respects rebase.autostash.  See
         # http://stackoverflow.com/a/30209750/125349
-        shell.call(["git", "rebase", "FETCH_HEAD"], echo=True)
+        #shell.call(["git", "rebase", "FETCH_HEAD"], echo=True)
```
Comment (gribozavr) [inline utils/update-checkout]: Debugging change?

File: utils/update-checkout [hunk 2]
```diff
@@ -217,17 +229,30 @@ By default, updates your checkouts of Swift, SourceKit, LLDB, and SwiftPM.""")
         "--config",
         default=os.path.join(SCRIPT_DIR, "update-checkout-config.json"),
         help="Configuration file to use")
+    parser.add_argument(
+        "--github-comment",
+        help='Use github comment to parse pull request for other repository.',
```
Comment (gribozavr) [inline utils/update-checkout]: "Check out related pull requests referenced in the given free-form GitHub-style comment." ?

File: utils/update-checkout [hunk 3]
```diff
@@ -217,17 +229,30 @@ By default, updates your checkouts of Swift, SourceKit, LLDB, and SwiftPM.""")
...
     scheme = args.scheme
+    github_comment = args.github_comment
 
     with open(args.config) as f:
         config = json.load(f)
     validate_config(config)
 
+    cross_repos_pr = {}
+    if github_comment:
+        print(github_comment)
```
Comment (gribozavr) [inline utils/update-checkout]: Debug print?

**Comment by shahmishal** [comment]
@swift-ci Python lint

**Comment by shahmishal** [comment]
@swift-ci Python lint

**Comment by shahmishal** [comment]
@swift-ci Python lint

**Comment by shahmishal** [comment]
@swift-ci Python lint

**Comment by shahmishal** [comment]
```
+ ./utils/python-lint
./utils/resolve-crashes.py:19:80: E501 line too long (94 > 79 characters)
```
Not related to my changes. This might be due to 1b11d9d4c3d5f1986146b481d64880310cb7ea24

File: utils/update-checkout (hunk 2 above)
Comment (shahmishal) [inline utils/update-checkout]: Fixed, thanks!

File: utils/update-checkout (hunk 3 above)
Comment (shahmishal) [inline utils/update-checkout]: Fixed Thanks!

**Comment by shahmishal** [comment]
@swift-ci please smoke test

**Comment by shahmishal** [comment]
@swift-ci please smoke test

**Comment by najacque** [comment]
@swift-ci Please ignore

Explanation: This is just a test

Scope: Just testing!

Issue: rdar://problem/28121222

Reviewed by: @shahmishal 

Risk: Low.

Testing: Test all the things!

**Pull request merged**